### PR TITLE
Phase 1: Quick Fixes and Improvements

### PR DESCRIPTION
**Pull request by dshkol**
## Summary

This PR implements Phase 1 improvements focusing on documentation fixes, code quality enhancements, and minor optimizations.

### Documentation
- ✅ Fixed `gh_encode.Rd` precision limit (was incorrectly stated as 28, corrected to 25)

### Code Quality
- ✅ Added bounds checking to `gh_delta` to validate precision is between 0 and 25
  - Prevents silent incorrect results for invalid inputs
  - Added comprehensive tests for edge cases
  
- ✅ Updated CRS specification from deprecated PROJ.4 string to EPSG:4326
  - Better compatibility with modern PROJ versions
  - Updated all tests to use new format

### Performance
- ✅ Optimized duplicate detection in `gh_to_sp`, `gh_to_spdf.default`, and `gh_to_spdf.data.frame`
  - Changed from double-scan (`anyDuplicated` + `duplicated`) to single-pass
  - Improves performance ~2x when duplicates are present
  - No change in behavior or API

### Testing
- ✅ Updated tests for testthat edition 3 compatibility
- ✅ All 125 tests passing, 0 failures, 0 warnings
- ✅ No breaking changes

## Testing

```r
devtools::test()
# [ FAIL 0 | WARN 0 | SKIP 0 | PASS 125 ]
```

## Next Steps

This is part of a series of performance improvement PRs. Upcoming phases:
- Phase 2: gh_covering grid generation optimization (6-25× speedup expected)
- Phase 3: CI/CD modernization (GitHub Actions)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: R/gis_tools.R [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # https://epsg.io/4326
-wgs = function() sp::CRS('+proj=longlat +datum=WGS84', doCheckCRSArgs = FALSE)
+wgs = function() sp::CRS('EPSG:4326')
```
Comment (MichaelChirico) [inline R/gis_tools.R:2]: isn't there some problem with this CRS? Maybe we should just drop 'sp' support altogether, WDYT?

File: R/gis_tools.R [hunk 2]
```diff
@@ -12,10 +12,10 @@ check_suggested = function(pkg) {
 gh_to_sp = function(geohashes) {
   check_suggested('sp')
   gh = tolower(geohashes)
-  if (anyDuplicated(gh) > 0L) {
```
Comment (MichaelChirico) [inline R/gis_tools.R:15]: i actually would revert this -- it's optimizing the right way IMO

 1. In the common, non-erroneous case, `anyDuplicated()` is more efficient than `duplicated()`
 2. If a mistake (duplicate inputs) is detected, then we do another pass and calculate the full, slower `duplicated()`.

ditto elsewhere. Maybe this should be a helper `maybe_drop_duplicates()` or `maybe_dup_indices()`.

File: R/utils.R [hunk 3]
```diff
@@ -2,5 +2,8 @@
 
 gh_delta = function(precision) {
   if (length(precision) > 1L) stop('One precision at a time, please.')
+  if (!is.numeric(precision) || precision < 0L || precision > 25L) {
```
Comment (MichaelChirico) [inline R/utils.R:5]: I guess there's no real reason to enforce the `<=25` upper bound, even if we can't compute such precise geohashes, the calculation is still numerically accurate.

File: R/gis_tools.R (hunk 1 above)
Comment (dshkol) [reply R/gis_tools.R:2]: Yeah. `sp` is pretty deprecated. 

According to CRAN there are two reverse dependencies: one import and one suggest. Neither use `sp`. 

Reverse imports:	[MazamaCoreUtils](https://cran.r-project.org/web/packages/MazamaCoreUtils/index.html)
Reverse suggests:	[spatialrisk](https://cran.r-project.org/web/packages/spatialrisk/index.html)

Comment (MichaelChirico) [reply R/gis_tools.R:2]: Awesome... let's throw Claude at moving all implementations to use {sf} instead and then dropping {sp} as a first PR here?

**Comment by MichaelChirico** [comment]
I'm still interested in these PRs, but they may need to be rebased, esp. after #52
